test/e2e/monitoring_stack_controller_test.go

```diff
@@ -594,3 +608,175 @@ func waitForStackDeletion(name string) error {
 		return errors.IsNotFound(err), nil
 	})
 }
+
+func namespaceSelectorTest(t *testing.T) {
+	// as a convention
+	// add labels to ns to indicate the stack responsible for monitoring the ns
+	// resourceSelector uses both stack and an app label
+	stackName := "multi-ns"
+	nsLabels := map[string]string{"monitoring.rhobs/stack": stackName}
+	resourceLabels := map[string]string{
+		"monitoring.rhobs/stack": stackName,
+		"app":                    "demo",
+	}
+
+	ms := newMonitoringStack(t, stackName,
+		msResourceSelector(resourceLabels),
+		msNamespaceSelector(nsLabels))
+
+	err := f.K8sClient.Create(context.Background(), ms)
+	assert.NilError(t, err, "failed to create a monitoring stack")
+
+	namespaces := []string{"test-ns-1", "test-ns-2", "test-ns-3"}
+
+	for _, ns := range namespaces {
+		err := deployDemoApp(t, ns, nsLabels, resourceLabels)
+		assert.NilError(t, err, "%s: deploying demo app failed", ns)
+	}
+
+	stopChan := make(chan struct{})
+	defer close(stopChan)
+	if pollErr := wait.Poll(5*time.Second, 2*time.Minute, func() (bool, error) {
+		err := f.StartServicePortForward(ms.Name+"-prometheus", e2eTestNamespace, "9090", stopChan)
+		return err == nil, nil
+	}); pollErr != nil {
+		t.Fatal(pollErr)
+	}
+
+	promClient := framework.NewPrometheusClient("http://localhost:9090")
+	if pollErr := wait.Poll(5*time.Second, 5*time.Minute, func() (bool, error) {
+		query := `prometheus_build_info{namespace=~"test-ns-.*"}`
+		result, err := promClient.Query(query)
+		if err != nil {
+			return false, nil
+		}
+
+		if len(result.Data.Result) != len(namespaces) {
+			return false, nil
+		}
+
+		return true, nil
+	}); pollErr != nil {
+		t.Fatal(pollErr)
+	}
+}
+
+func deployDemoApp(t *testing.T, nsName string, nsLabels, resourceLabels map[string]string) error {
+
+	// now deploy a prometheus instance into test-ns
```

Comment:
nit: here we are just creating the namespace, I feel like if we want to go with this comment it should be a comment on the function, no?

Reply:
or delete the comment totally since this was refactored and the comment made sense in a loop.